compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/Desugar.java

```diff
@@ -1914,6 +2067,43 @@ public void visit(BLangSimpleVarRef varRefExpr) {
         result = addConversionExprIfRequired(genVarRefExpr, targetType);
     }
 
+    private BVarSymbol resolveEnvMapSymbol(SymbolEnv env, BVarSymbol varSymbol) {
+        BVarSymbol bVarSymbol = null;
+        while (env.node.getKind() != NodeKind.PACKAGE) {
+            if (env.exposedClosureHolder.closuresExposed.contains(varSymbol)) {
+                bVarSymbol = env.exposedClosureHolder.mapSymbol;
+                break;
+            }
+            env = env.enclEnv;
+        }
+        return bVarSymbol;
+    }
+
+    private boolean updateClosureVars(BLangSimpleVarRef varRefExpr, BVarSymbol mapSymbol) {
+        // Get type of the index based access expression
+        BType typeOfExpr = isBasicType(varRefExpr.type) ? ((BMapType) mapSymbol.type).constraint : varRefExpr.type;
+        // Create the index based access expression
+        BLangLiteral indexExpr = ASTBuilderUtil.createLiteral(varRefExpr.pos, symTable.stringType,
+                varRefExpr.variableName.value);
+        BLangIndexBasedAccess accessExpr = ASTBuilderUtil.createIndexBasesAccessExpr(varRefExpr.pos, typeOfExpr,
+                mapSymbol, indexExpr);
+        // If its in the LHS of an assignment
+        if (varRefExpr.lhsVar) {
+            // x = 1 ==> $innerMap$1["x"] = <any> 1
+            result = rewriteExpr(accessExpr);
+            return true;
+        }
+        // int z = x + 1 ==> int z = <int>$innerMap$1["x"] + 1;
+        result = rewriteExpr(addConversionExprIfRequired(accessExpr, varRefExpr.type));
+        return true;
+    }
+
+    private boolean isBasicType(BType bType) {
```

Comment:
Is this method used to check whether a type is a simple basic type? If that is the case, shall we use a simple `<=` boolean operator to check whether it is a simple basic type? Also, if this check is for simple basic type shall we modify the method name as `isSimpleBasicType`?

i.e: `bType.tag <= TypeTags.BOOLEAN_TAG`